src/reuse/header.py

```diff
@@ -205,16 +206,35 @@ def _extract_shebang(prefix: str, text: str) -> tuple[str, str]:
     tuple of (shebang, reduced_text).
     """
     shebang_lines = []
-    for line in text.splitlines():
+    for line in text.splitlines(keepends=True):
         if line.startswith(prefix):
             shebang_lines.append(line)
             text = text.replace(line, "", 1)
         else:
             break
-    shebang = "\n".join(shebang_lines)
+    shebang = "".join(shebang_lines)
     return (shebang, text)
 
 
+def place_header(
+    header: str, before: str, after: str, has_existing_header: bool
+) -> str:
+    """Combine the header and the text from the file into the final output"""
+    new_text = f"{header}\n"
+    if before.strip():
+        new_text = f"{before.rstrip()}\n\n{new_text}"
+    if after.strip():
+        # Create space between header and following code only if a newline
+        # doesn't already exist, and there wasn't previously a header.
+        if not (has_existing_header or after.startswith("\n")):
```

Comment:
```suggestion
        if not has_existing_header and not after.startswith("\n"):
```

This is more easily parsed by humans.